### ISSUE TITLE
[TorchScipt] Custom Objects C++ API

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#49261]
## 🐛 Bug: Creating TorchScript Objects in C++

@samster25  pointed out in a presentation that there is no public  API for creating a User-defined Dataclass or named tuple in the C++ API. 

More generally, the C++ API is hard to use. In the past, we have considered having a separate set of user-exposed IValue bindings. We could also consider having a tutorial for doing common things in JIT documentation. 

cc @SplitInfinity who has been working on improving custom classes support

**Comment by glaringlee** [comment]
Not sure whether this should be labeled as `jit`, please change the label if this is not jit related. Thanks.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
